modules/k3s/k3s.go

```diff
@@ -72,6 +77,31 @@ func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomize
 	return &K3sContainer{Container: container}, nil
 }
 
+func getContainerHost(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (string, error) {
+	// Use a dummy request to get the provider from options.
+	var req testcontainers.GenericContainerRequest
+	for _, opt := range opts {
+		opt.Customize(&req)
+	}
+
+	logging := req.Logger
+	if logging == nil {
+		logging = testcontainers.Logger
+	}
+	p, err := req.ProviderType.GetProvider(testcontainers.WithLogger(logging))
```

Comment:
I'm taking a look to check if it's possible to get a Provider instance from the library, instead of from the container request + customisers. In fact in https://github.com/testcontainers/testcontainers-go/pull/1251/files#diff-e9b78e9b34e14f076ba250ebd1042b315e63673569f82372a8064cece5f034bfR25 I'm deprecating the field in order to get them automatically from the environment.

In any case, I think this is good enough for now :)